pinc/Project.inc

```diff
@@ -465,6 +466,39 @@ EOS;
         return $hold_states;
     }
 
+    public function add_holds($states)
+    {
+        global $pguser;
+
+        $values = [];
+        foreach ($states as $state)
+        {
+            $values[] = "('$this->projectid', '$state')";
+        }
+        $values = join(', ', $values);
+
+        mysqli_report(MYSQLI_REPORT_ERROR | MYSQLI_REPORT_STRICT);
```

Comment:
Based on our conversation in the other MR, until we vet the code and understand if subsequent `mysqli` calls can handle exceptions being thrown on failures, we should stick with the current convention of `mysqli_query(...) or die("error")` instead of using this to make errors turn into raised exceptions.

I think what you have is the direction we want to go, but I want it to be in `base.inc` for all calls everywhere after we make sure it doesn't break anything.